### ISSUE TITLE
Configurable classname/name switch

### ISSUE THREAD

**Issue by clouless** [michaelleeallen/mocha-junit-reporter#51]
I would like to contribute a PR.
But first we may need to talk about how you would like to have it.

**TL;DR** Make `classname` and `name` switchable via config

&nbsp;

### Problem

Regarding `<testcase name="" classname=""` entries:

  * Since I use junit.xml results generated by java-surefire together with the ones generated by this reporter I want **`classname` to be switched with `name`**.

Example from Java:

```xml
<?xml version="1.0" encoding="UTF-8" standalone="no"?>
<testsuites failures="0" name="Custom" tests="14" time="0.8200000000000001">
  <testsuite errors="0" failures="0"
             name="ut.io.codeclou.customfield.editor.MyComponentUnitTest"
             skipped="0" tests="1" time="0">
    <testcase
      time="0.359"
      classname="ut.io.codeclou.customfield.editor.rest.CustomFieldHelperTest" 
      name="testReorderOptionsMoveTopMiddleToBottom"/>
    <testcase
      time="0.018"
      classname="ut.io.codeclou.customfield.editor.rest.CustomFieldHelperTest" 
      name="testDoBothListsHaveCommonValue"/>
  </testsuite>
</testsuites>
```

As you can see `classname` is really used for the className and `name` for the testMethodName.

The mocha-junit-reporter generates it the other way. When I say

```js
describe('Global Permissions Test', () => {
  it('should be possible for admin to lookup bob by using his lastname in userSearch', (done) => {
   ...
  })
})
```

It generates

```xml
...
  <testsuite name="Global Permissions Test" timestamp="2017-04-08T07:04:13" tests="3"    
             file="/work-private/tests/rest-api/1/test-local.js" failures="0" time="0.244">
    <testcase
      time="0.146"
      classname="should be possible for admin to lookup bob by using his lastname in userSearch"
      name="suiteName a.s.o">
    </testcase>
  </testsuite>
...
```

And I want it to look like:

```xml
...
  <testsuite name="Global Permissions Test" timestamp="2017-04-08T07:04:13" tests="3"    
             file="/work-private/tests/rest-api/1/test-local.js" failures="0" time="0.244">
    <testcase
      time="0.146"
      classname="suiteName a.s.o"
      name="should be possible for admin to lookup bob by using his lastname in userSearch">
    </testcase>
  </testsuite>
...
```

&nbsp;

### Proposal

Add `reporterOptions` property named `testCaseSwitchClassnameWithName` that defaults to `false`.

So when you call:

```
TEST_CASE_SWITCH_CLASSNAME_WITH_NAME=true mocha test --reporter mocha-junit-reporter
```

Then it would generate it with switched name/classname for all testcase-entries.

&nbsp;

What do you say?

**Comment by clayreimann** [comment]
That seems like a reasonable request, feel free to PR away.

My only implementation note is that we've been moving from ENV_VARs to adding options via the reporter options, so prefer adding more reporter options to reading ENV_VARs

**Comment by clouless** [comment]
cool, I will keep that in mind and provide a PR near the end of the week :)

**Comment by jjdonov** [comment]
Hey there, 

Happy to see this PR got merged, I'd love to use it. It seems that there have been some new features that have landed on master but have yet to make their way to npm. I was curious to know when that might happen?

**Comment by clayreimann** [comment]
My bad, I read the `.travis.yml` almost correctly. Pushed the new version so should hit npm shortly.
